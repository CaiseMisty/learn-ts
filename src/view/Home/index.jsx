import React from 'react';

export default class Home extends React.Component {
  static displayName = 'Home';

  render() {
    return <div>这里展示首页</div>;
  }
}
