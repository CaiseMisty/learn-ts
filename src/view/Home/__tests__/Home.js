import React from 'react';
import RsfApp from '@tencent/rsf';
import { ConfigRouters } from '@tencent/rsf';
import { MemoryRouter as Router } from 'react-router-dom';
import { mount, render } from 'enzyme';
import MainLayout from '$view/layout/Main';
import Home from '$view/Home';
import routesConfig from '$src/routes.config';
import mutations from '$src/mutations';

describe('src/view/Home', () => {
  function TestComponent() {
    return (
      <RsfApp
        mutations={mutations}
        initialState={{
          $loading: {
            '/globalData': false,
          },
        }}
      >
        <Router initialEntries={['/']}>
          <MainLayout>
            <ConfigRouters routes={routesConfig} />
          </MainLayout>
        </Router>
      </RsfApp>
    );
  }

  it('equals snapshots', () => {
    const html = render(<TestComponent />);
    expect(html).toMatchSnapshot();
  });

  it('has Home component`', () => {
    const wrapper = mount(<TestComponent />);
    expect(wrapper.find(Home)).toHaveLength(1);
  });

  it('Home component content`', () => {
    const wrapper = mount(<TestComponent />);
    expect(wrapper.find(Home).text()).toEqual('这里展示首页');
  });
});
