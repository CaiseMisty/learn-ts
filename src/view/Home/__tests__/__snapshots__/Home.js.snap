// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/view/Home equals snapshots 1`] = `
<section
  class="layoutContainer ant-layout"
>
  <aside
    class="ant-layout-sider ant-layout-sider-dark"
    style="flex:0 0 250px;max-width:250px;min-width:250px;width:250px"
  >
    <div
      class="ant-layout-sider-children"
    >
      <div
        class="logo"
      >
        <img
          alt=""
          src=""
        />
        <h3
          class="logoTitle"
        >
          广告数据分析平台
        </h3>
      </div>
      <ul
        class="ant-menu ant-menu-dark ant-menu-root ant-menu-inline"
        history="[object Object]"
        role="menu"
      >
        <li
          class="ant-menu-item"
          role="menuitem"
          style="padding-left:24px"
        >
          <a
            href="/"
          >
            <i
              aria-label="icon: home"
              class="anticon anticon-home"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="home"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 0 0-44.4 0L77.5 505a63.9 63.9 0 0 0-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0 0 18.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                />
              </svg>
            </i>
            <span>
              主页
            </span>
          </a>
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-inline"
          role="menuitem"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            style="padding-left:24px"
          >
            <span>
              <i
                aria-label="icon: user"
                class="anticon anticon-user"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="user"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                  />
                </svg>
              </i>
              <span>
                用户管理
              </span>
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
          <div />
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-inline"
          role="menuitem"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            style="padding-left:24px"
          >
            <span>
              <i
                aria-label="icon: warning"
                class="anticon anticon-warning"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="warning"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M464 720a48 48 0 1 0 96 0 48 48 0 1 0-96 0zm16-304v184c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V416c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8zm475.7 440l-416-720c-6.2-10.7-16.9-16-27.7-16s-21.6 5.3-27.7 16l-416 720C56 877.4 71.4 904 96 904h832c24.6 0 40-26.6 27.7-48zm-783.5-27.9L512 239.9l339.8 588.2H172.2z"
                  />
                </svg>
              </i>
              <span>
                异常页面
              </span>
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
          <div />
        </li>
      </ul>
    </div>
  </aside>
  <section
    class="main ant-layout"
  >
    <header
      class="header ant-layout-header"
    >
      <div
        class="headerLeft"
      >
        <i
          aria-label="icon: menu-fold"
          class="anticon anticon-menu-fold menuFoldBtn"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="menu-fold"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"
            />
          </svg>
        </i>
      </div>
      <div
        class="headerRight"
      >
        <span
          class="ant-avatar avatar ant-avatar-circle"
        >
          <span
            class="ant-avatar-string"
            style="opacity:0"
          />
        </span>
      </div>
    </header>
    <main
      class="body ant-layout-content"
    >
      <div
        class="breadcrumb ant-breadcrumb"
      >
        <span>
          <span
            class="ant-breadcrumb-link"
          >
            主页
          </span>
          <span
            class="ant-breadcrumb-separator"
          >
            /
          </span>
        </span>
      </div>
      <div
        class="bodyContent"
      >
        <div>
          这里展示首页
        </div>
      </div>
    </main>
    <footer
      class="footer ant-layout-footer"
    >
      RSF @2020
    </footer>
  </section>
</section>
`;
